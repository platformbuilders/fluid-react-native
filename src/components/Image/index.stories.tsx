import type { Meta } from '@storybook/react';
import Image from '.';

const meta: Meta<typeof Image> = {
  title: 'Image',
  component: Image,
  args: {
    source: { uri: 'https://thispersondoesnotexist.com', cache: 'cacheOnly' },
    accessibility: 'Image',
    animationColor: '#355deb',
    animationContainerColor: '#355deb',
    containerStyle: { width: 220, height: 220 },
    displayPlaceholder: true,
    id: 'image-id',
    resizeMode: 'cover',
    style: { width: 220, height: 220 },
    testID: 'image-test-id',
  },
  parameters: {
    notes:
      'Esse componente tem como função auxiliar na manipulação, manutenção e padronização de imagens dentro de uma aplicação.',
  },
};

export default meta;

export const Default = {};
