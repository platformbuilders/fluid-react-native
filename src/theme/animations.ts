import { Animations } from '@platformbuilders/theme-toolkit';
import {
  LoadDotsAnimation,
  LoadDotsContrastAnimation,
  LoadingCircleAnimation,
  LoadingCircleContrastAnimation,
  LoadingProgressAnimation,
} from '../assets/animations';

export default {
  circularLoading: LoadingCircleAnimation,
  contrastCircularLoading: LoadingCircleContrastAnimation,
  buttonLoading: LoadDotsAnimation,
  contrastButtonLoading: LoadDotsContrastAnimation,
  linearLoading: LoadingProgressAnimation,
  contrastLinearLoading: LoadingProgressAnimation,
} as Animations;
