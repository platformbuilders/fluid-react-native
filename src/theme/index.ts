export { default as colors } from './colors';
export { default as spacing } from './spacing';
export { default as radius } from './radius';
export { default as animations } from './animations';
export { default as typography } from './typography';
