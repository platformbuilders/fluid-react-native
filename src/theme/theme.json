{
  "spacing": {
    "min": {
      "value": 1,
      "type": "spacing"
    },
    "xxs": {
      "value": 2,
      "type": "spacing"
    },
    "xs": {
      "value": 4,
      "type": "spacing"
    },
    "sm": {
      "value": 8,
      "type": "spacing"
    },
    "md": {
      "value": 16,
      "type": "spacing"
    },
    "lg": {
      "value": 24,
      "type": "spacing"
    },
    "xl": {
      "value": 32,
      "type": "spacing"
    },
    "xxl": {
      "value": 48,
      "type": "spacing"
    },
    "max": {
      "value": 64,
      "type": "spacing"
    }
  },
  "sizing": {
    "min": {
      "value": 1,
      "type": "sizing"
    },
    "xxs": {
      "value": 2,
      "type": "sizing"
    },
    "xs": {
      "value": 4,
      "type": "sizing"
    },
    "sm": {
      "value": 8,
      "type": "sizing"
    },
    "md": {
      "value": 16,
      "type": "sizing"
    },
    "lg": {
      "value": 24,
      "type": "sizing"
    },
    "xl": {
      "value": 32,
      "type": "sizing"
    },
    "xxl": {
      "value": 48,
      "type": "sizing"
    },
    "max": {
      "value": 64,
      "type": "sizing"
    }
  },
  "borderRadius": {
    "min": {
      "value": 0,
      "type": "borderRadius"
    },
    "xxs": {
      "value": 1,
      "type": "borderRadius"
    },
    "xs": {
      "value": 2,
      "type": "borderRadius"
    },
    "sm": {
      "value": 4,
      "type": "borderRadius"
    },
    "md": {
      "value": 6,
      "type": "borderRadius"
    },
    "lg": {
      "value": 8,
      "type": "borderRadius"
    },
    "xl": {
      "value": 16,
      "type": "borderRadius"
    },
    "xxl": {
      "value": 32,
      "type": "borderRadius"
    },
    "max": {
      "value": 999,
      "type": "borderRadius"
    }
  },
  "borderWidth": {
    "min": {
      "value": 0.5,
      "type": "borderWidth"
    },
    "xxs": {
      "value": 1,
      "type": "borderWidth"
    },
    "xs": {
      "value": 1.5,
      "type": "borderWidth"
    },
    "sm": {
      "value": 2,
      "type": "borderWidth"
    },
    "md": {
      "value": 3,
      "type": "borderWidth"
    },
    "lg": {
      "value": 4,
      "type": "borderWidth"
    },
    "xl": {
      "value": 6,
      "type": "borderWidth"
    },
    "xxl": {
      "value": 8,
      "type": "borderWidth"
    },
    "max": {
      "value": 12,
      "type": "borderWidth"
    }
  },
  "boxShadow": {
    "z0": {
      "value": {
        "x": 0,
        "y": 0,
        "spread": 0,
        "color": "",
        "blur": 0
      },
      "type": "boxShadow"
    },
    "z1": {
      "value": {
        "x": 0,
        "y": 1,
        "spread": 0,
        "color": "rgba(0, 0, 0, 0.1)",
        "blur": 3
      },
      "type": "boxShadow"
    },
    "z2": {
      "value": {
        "x": 0,
        "y": 4,
        "spread": -1,
        "color": "rgba(0, 0, 0, 0.1)",
        "blur": 6
      },
      "type": "boxShadow"
    },
    "z3": {
      "value": {
        "x": 0,
        "y": 10,
        "spread": -3,
        "color": "rgba(0, 0, 0, 0.1)",
        "blur": 15
      },
      "type": "boxShadow"
    },
    "z4": {
      "value": {
        "x": 0,
        "y": 20,
        "spread": -5,
        "color": "rgba(0, 0, 0, 0.1)",
        "blur": 25
      },
      "type": "boxShadow"
    },
    "z5": {
      "value": {
        "x": 0,
        "y": 25,
        "spread": -12,
        "color": "rgba(0, 0, 0, 0.2)",
        "blur": 50
      },
      "type": "boxShadow"
    }
  },
  "brand": {
    "primary": {
      "light": {
        "value": "#7684F9",
        "type": "color"
      },
      "main": {
        "value": "#4255EF",
        "type": "color"
      },
      "dark": {
        "value": "#1B288E",
        "type": "color"
      },
      "contrast": {
        "value": "#ffffff",
        "type": "color"
      }
    },
    "secondary": {
      "light": {
        "value": "#8BD6F8",
        "type": "color"
      },
      "main": {
        "value": "#42ADEF",
        "type": "color"
      },
      "dark": {
        "value": "#337BB7",
        "type": "color"
      },
      "contrast": {
        "value": "#ffffff",
        "type": "color"
      }
    },
    "tertiary": {
      "light": {
        "value": "#BA96F5",
        "type": "color"
      },
      "main": {
        "value": "#8442EF",
        "type": "color"
      },
      "dark": {
        "value": "#4A00DC",
        "type": "color"
      },
      "contrast": {
        "value": "#ffffff",
        "type": "color"
      }
    },
    "accent": {
      "light": {
        "value": "#FBD268",
        "type": "color"
      },
      "main": {
        "value": "#F49D19",
        "type": "color"
      },
      "dark": {
        "value": "#A1530A",
        "type": "color"
      },
      "contrast": {
        "value": "#351B02",
        "type": "color"
      }
    }
  },
  "success": {
    "light": {
      "value": "#F4FCD1",
      "type": "color"
    },
    "main": {
      "value": "#96C920",
      "type": "color"
    },
    "dark": {
      "value": "#396006",
      "type": "color"
    },
    "contrast": {
      "value": "#ffffff",
      "type": "color"
    }
  },
  "danger": {
    "light": {
      "value": "#FFE8D8",
      "type": "color",
      "description": "the 100 shade of a color"
    },
    "main": {
      "value": "#FF4F3F",
      "type": "color"
    },
    "dark": {
      "value": "#7A0C27",
      "type": "color",
      "description": "the 900s shade of a color"
    },
    "contrast": {
      "value": "#ffffff",
      "type": "color"
    }
  },
  "warning": {
    "light": {
      "value": "#FFF8CE",
      "type": "color"
    },
    "main": {
      "value": "#FFCA0C",
      "type": "color"
    },
    "dark": {
      "value": "#7A5502",
      "type": "color"
    },
    "contrast": {
      "value": "#261B00",
      "type": "color"
    }
  },
  "info": {
    "light": {
      "value": "#C2ECEC",
      "type": "color"
    },
    "main": {
      "value": "#42CCD4",
      "type": "color"
    },
    "dark": {
      "value": "#186F82",
      "type": "color"
    },
    "contrast": {
      "value": "#ffffff",
      "type": "color"
    }
  },
  "text": {
    "main": {
      "value": "#444444",
      "type": "color"
    },
    "dark": {
      "value": "#000",
      "type": "color"
    },
    "light": {
      "value": "#999999",
      "type": "color"
    },
    "contrast": {
      "value": "#fff",
      "type": "color"
    }
  },
  "background": {
    "z0": {
      "value": "#ffffff",
      "type": "color"
    },
    "z1": {
      "value": "#fafafa",
      "type": "color"
    },
    "z2": {
      "value": "#f0f0f0",
      "type": "color"
    },
    "z3": {
      "value": "#dddddd",
      "type": "color"
    },
    "z4": {
      "value": "#d1d1d1",
      "type": "color"
    }
  },
  "opacity": {
    "disabled": {
      "value": "30%",
      "type": "opacity"
    },
    "empty": {
      "value": "50%",
      "type": "opacity"
    },
    "overlay": {
      "value": "80%",
      "type": "opacity"
    }
  },
  "fontFamilies": {
    "heading": {
      "value": "Roboto",
      "type": "fontFamilies"
    },
    "body": {
      "value": "Roboto",
      "type": "fontFamilies"
    }
  },
  "lineHeights": {
    "heading": {
      "value": "110%",
      "type": "lineHeights"
    },
    "body": {
      "value": "140%",
      "type": "lineHeights"
    }
  },
  "letterSpacing": {
    "default": {
      "value": 0,
      "type": "letterSpacing"
    },
    "bigger": {
      "value": "105%",
      "type": "letterSpacing"
    }
  },
  "paragraphSpacing": {
    "h1": {
      "value": 32,
      "type": "paragraphSpacing"
    },
    "h2": {
      "value": 26,
      "type": "paragraphSpacing"
    }
  },
  "fontWeights": {
    "headingRegular": {
      "value": "Regular",
      "type": "fontWeights"
    },
    "headingBold": {
      "value": "Bold",
      "type": "fontWeights"
    },
    "bodyRegular": {
      "value": "Regular",
      "type": "fontWeights"
    },
    "bodyBold": {
      "value": "Bold",
      "type": "fontWeights"
    }
  },
  "fontSizes": {
    "min": {
      "value": 10,
      "type": "fontSizes"
    },
    "xxs": {
      "value": 11,
      "type": "fontSizes"
    },
    "xs": {
      "value": 13,
      "type": "fontSizes"
    },
    "sm": {
      "value": 14,
      "type": "fontSizes"
    },
    "md": {
      "value": 15,
      "type": "fontSizes"
    },
    "lg": {
      "value": 17,
      "type": "fontSizes"
    },
    "xl": {
      "value": 20,
      "type": "fontSizes"
    },
    "xxl": {
      "value": 24,
      "type": "fontSizes"
    },
    "max": {
      "value": 32,
      "type": "fontSizes"
    }
  },
  "themeRadius": {
    "button": {
      "value": 3,
      "type": "borderRadius"
    },
    "card": {
      "value": 6,
      "type": "borderRadius"
    },
    "input": {
      "value": 3,
      "type": "borderRadius"
    }
  },
  "typography": {
    "H1": {
      "Bold": {
        "type": "typography",
        "value": {
          "fontFamily": "$fontFamilies.heading",
          "fontWeight": "$fontWeights.headingBold",
          "lineHeight": "$lineHeights.heading",
          "fontSize": "$fontSizes.h1",
          "paragraphSpacing": "$paragraphSpacing.h1",
          "letterSpacing": "$letterSpacing.decreased"
        }
      },
      "Regular": {
        "type": "typography",
        "value": {
          "fontFamily": "$fontFamilies.heading",
          "fontWeight": "$fontWeights.headingRegular",
          "lineHeight": "$lineHeights.heading",
          "fontSize": "$fontSizes.h1",
          "paragraphSpacing": "$paragraphSpacing.h1",
          "letterSpacing": "$letterSpacing.decreased"
        }
      }
    },
    "H2": {
      "Bold": {
        "type": "typography",
        "value": {
          "fontFamily": "$fontFamilies.heading",
          "fontWeight": "$fontWeights.headingBold",
          "lineHeight": "$lineHeights.heading",
          "fontSize": "$fontSizes.h2",
          "paragraphSpacing": "$paragraphSpacing.h2",
          "letterSpacing": "$letterSpacing.decreased"
        }
      },
      "Regular": {
        "type": "typography",
        "value": {
          "fontFamily": "$fontFamilies.heading",
          "fontWeight": "$fontWeights.headingRegular",
          "lineHeight": "$lineHeights.heading",
          "fontSize": "$fontSizes.h2",
          "paragraphSpacing": "$paragraphSpacing.h2",
          "letterSpacing": "$letterSpacing.decreased"
        }
      }
    },
    "Body": {
      "type": "typography",
      "value": {
        "fontFamily": "$fontFamilies.body",
        "fontWeight": "$fontWeights.bodyRegular",
        "lineHeight": "$lineHeights.heading",
        "fontSize": "$fontSizes.body",
        "paragraphSpacing": "$paragraphSpacing.h2"
      }
    }
  }
}
