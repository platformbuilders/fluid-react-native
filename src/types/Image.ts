import { StyleProp, ViewStyle } from 'react-native';
import { FastImageProps } from 'react-native-fast-image';

export interface ImageProps extends FastImageProps {
  id?: string;
  accessibility?: string;
  displayPlaceholder?: boolean;
  containerStyle?: any;
  animationColor?: string;
  animationContainerColor?: string;
}
