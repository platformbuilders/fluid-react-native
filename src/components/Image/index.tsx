import React, { useContext, useState } from 'react';
import FastImage from 'react-native-fast-image';
import { Placeholder, PlaceholderMedia, Shine } from 'rn-placeholder';
import { ThemeContext } from 'styled-components';
import { ImageProps } from '../../types';
import { Wrapper } from './styles';

const Image: React.FC<ImageProps> = ({
  id,
  accessibility,
  containerStyle,
  displayPlaceholder = true,
  animationColor,
  animationContainerColor,
  style,
  ...rest
}) => {
  const [isLoading, setIsLoading] = useState(true);
  const theme = useContext(ThemeContext);

  const onImageLoad = () => {
    console.log('LOG:  onImageLoad  onImageLoad:');
    setIsLoading(false);
  };

  const onImageLoadStart = () => {
    console.log('LOG:  onImageLoadStart  onImageLoadStart:');
    setIsLoading(true);
  };

  const animationStyling = (animationProps: any) => (
    <Shine
      {...animationProps}
      style={[
        animationProps.style,
        {
          backgroundColor: animationColor || theme.background.z3,
        },
      ]}
    />
  );

  return (
    <Wrapper style={containerStyle}>
      {isLoading && displayPlaceholder && (
        <Placeholder Animation={animationStyling}>
          <PlaceholderMedia
            style={[
              style,
              {
                backgroundColor: animationContainerColor || theme.background.z2,
              },
            ]}
          />
        </Placeholder>
      )}
      <FastImage
        testID={id}
        accessibilityLabel={accessibility}
        onLoadStart={onImageLoadStart}
        onLoad={onImageLoad}
        style={style}
        {...rest}
      />
    </Wrapper>
  );
};

export default Image;
