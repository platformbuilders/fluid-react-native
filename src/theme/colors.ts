import { darken, lighten } from 'polished';
import { Colors } from '../types';

const primary = '#C6B09E';
const secondary = '#805281';
const tertiary = '#2A2E39';
const accent = '#547E6A';
const disabled = '#eeeeee';

const colors: Colors = {
  text: '#212121',
  info: '#4096D1',
  success: '#357a38',
  warning: '#F5B800',
  failure: '#cc0000',
  disabled: {
    light: lighten(0.05, disabled),
    main: disabled,
    dark: darken(0.3, disabled),
    contrast: '#cccccc',
  },
  primary: {
    light: lighten(0.05, primary),
    main: primary,
    dark: darken(0.12, primary),
    contrast: '#ffffff',
  },
  secondary: {
    light: lighten(0.05, secondary),
    main: secondary,
    dark: darken(0.1, secondary),
    contrast: '#ffffff',
  },
  tertiary: {
    light: lighten(0.15, tertiary),
    main: tertiary,
    dark: darken(0.1, tertiary),
    contrast: '#ffffff',
  },
  accent: {
    light: lighten(0.05, accent),
    main: accent,
    dark: darken(0.05, accent),
    contrast: '#ffffff',
  },
};

export default colors;
