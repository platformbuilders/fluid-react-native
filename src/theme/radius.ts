import { moderateScale } from 'react-native-size-matters';
import { Radius } from '../types';

export default {
  smallRadius: `${moderateScale(3)}px`,
  mediumRadius: `${moderateScale(8)}px`,
  largeRadius: `${moderateScale(15)}px`,
  modalRadius: `${moderateScale(20)}px`,
  buttonRadius: `${moderateScale(7)}px`,
} as Radius;
