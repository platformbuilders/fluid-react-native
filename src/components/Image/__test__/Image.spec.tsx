import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components/native';
import Image from '..';
import theme from '../../../theme';

const uri =
  'https://www.google.com/images/branding/googlelogo/2x/googlelogo_color_272x92dp.png';

jest.mock('rn-placeholder', () => {
  const RealPlaceholder = jest.requireActual('rn-placeholder');

  const MockShine = jest.fn().mockImplementation((props) => {
    // Mock para o componente Shine
    return <RealPlaceholder.Shine {...props} />;
  });

  return {
    ...RealPlaceholder,
    Shine: MockShine,
  };
});

describe('<Image />', () => {
  it('should render Image component correctly', () => {
    const render = renderer.create(
      <ThemeProvider theme={theme}>
        <Image source={{ uri }} id="testing" accessibility="testing_image" />
      </ThemeProvider>,
    );

    expect(render.toJSON()).toMatchSnapshot();
  });

  it('should render with no placeholder', () => {
    const render = renderer.create(
      <ThemeProvider theme={theme}>
        <Image
          id="testing"
          accessibility="testing_image"
          displayPlaceholder={false}
          source={{ uri }}
        />
      </ThemeProvider>,
    );

    expect(render.toJSON()).toMatchSnapshot();
  });

  it('should render with custom placeholder animation color', () => {
    const render = renderer.create(
      <ThemeProvider theme={theme}>
        <Image
          id="testing"
          accessibility="testing_image"
          animationColor="red"
          source={{ uri }}
        />
      </ThemeProvider>,
    );

    expect(render.toJSON()).toMatchSnapshot();
  });

  it('should render with custom placeholder animation and container colors', () => {
    const render = renderer.create(
      <ThemeProvider theme={theme}>
        <Image
          id="testing"
          accessibility="testing_image"
          animationColor="red"
          animationContainerColor="blue"
          source={{ uri }}
        />
      </ThemeProvider>,
    );

    expect(render.toJSON()).toMatchSnapshot();
  });
});
