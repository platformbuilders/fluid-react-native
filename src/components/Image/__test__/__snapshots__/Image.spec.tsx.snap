// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> should render Image component correctly 1`] = `
<View
  style={
    [
      {},
    ]
  }
>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <Image
      accessibilityLabel="testing_image"
      onLoad={[Function]}
      onLoadStart={[Function]}
      resizeMode="cover"
      source={
        {
          "uri": "https://www.google.com/images/branding/googlelogo/2x/googlelogo_color_272x92dp.png",
        }
      }
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "tintColor": undefined,
          },
        ]
      }
      testID="testing"
    />
  </View>
</View>
`;

exports[`<Image /> should render with custom placeholder animation and container colors 1`] = `
<View
  style={
    [
      {},
    ]
  }
>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <Image
      accessibilityLabel="testing_image"
      onLoad={[Function]}
      onLoadStart={[Function]}
      resizeMode="cover"
      source={
        {
          "uri": "https://www.google.com/images/branding/googlelogo/2x/googlelogo_color_272x92dp.png",
        }
      }
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "tintColor": undefined,
          },
        ]
      }
      testID="testing"
    />
  </View>
</View>
`;

exports[`<Image /> should render with custom placeholder animation color 1`] = `
<View
  style={
    [
      {},
    ]
  }
>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <Image
      accessibilityLabel="testing_image"
      onLoad={[Function]}
      onLoadStart={[Function]}
      resizeMode="cover"
      source={
        {
          "uri": "https://www.google.com/images/branding/googlelogo/2x/googlelogo_color_272x92dp.png",
        }
      }
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "tintColor": undefined,
          },
        ]
      }
      testID="testing"
    />
  </View>
</View>
`;

exports[`<Image /> should render with no placeholder 1`] = `
<View
  style={
    [
      {},
    ]
  }
>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <Image
      accessibilityLabel="testing_image"
      onLoad={[Function]}
      onLoadStart={[Function]}
      resizeMode="cover"
      source={
        {
          "uri": "https://www.google.com/images/branding/googlelogo/2x/googlelogo_color_272x92dp.png",
        }
      }
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "tintColor": undefined,
          },
        ]
      }
      testID="testing"
    />
  </View>
</View>
`;
